### PR TITLE
[doc][hackathon] To add Adam Optimizer to the documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of Adam Algorithm to the documentation.  For more details, we refer to the paper  https://arxiv.org/abs/1412.6980

Screen Shot 2021-08-27 at 6 37 54 PM

cc @vincentqb @iramazanli

### CODE REVIEW

File: torch/optim/adam.py [hunk 1]
```diff
@@ -25,8 +53,6 @@ class Adam(Optimizer):
 
     .. _Adam\: A Method for Stochastic Optimization:
         https://arxiv.org/abs/1412.6980
-    .. _Decoupled Weight Decay Regularization:
```
Comment (albanD) [inline torch/optim/adam.py:28]: This is not true anymore?

**Review by albanD — APPROVED** [review]
LGTM

File: torch/optim/adam.py (hunk 1 above)
Comment (iramazanli) [reply torch/optim/adam.py:28]: This paper contains AdamW rather than Adam, that's why i deleted this :) thanks for pointing it out.

Comment (albanD) [reply torch/optim/adam.py:28]: ok

**Pull request closed without merging**